src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTORY = 0.75F;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        putNode(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getIndex(key);
+        Node<K, V> currentNode = table[hash];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putNode(K key, V value) {
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode.next != null) {
+                if (Objects.equals(key, currentNode.key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            if (Objects.equals(key, currentNode.key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode.next = newNode;
+        }
+        size++;
+    }
+
+    private int getIndex(final K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private void resize() {
+        if (size >= table.length * LOAD_FACTORY) {
+            size = 0;
+            int newCapacity = table.length << 1;
+            Node<K, V>[] tmpTable = table;
+            table = new Node[newCapacity];
+            for (Node<K, V> node : tmpTable) {
+                if (node != null) {
+                    putNode(node.key, node.value);
+                    Node<K, V> currentNode = node;
+                    while (currentNode.next != null) {
+                        putNode(currentNode.next.key, currentNode.next.value);
+                        currentNode = currentNode.next;
+                    }
```

Comment:
you can optimize your solution. `while` can check your null, and you don't need additional `if` before 
```suggestion
                while (node != null) {
                    putNode(node.key, node.value);
                    node = node.next;
                }
```